### PR TITLE
Change timestamp alias to timestamp without time zone

### PR DESCRIPTION
**Pull request by mkleen**
## Summary of the changes / Why this improves CrateDB

Resolves https://github.com/crate/crate/issues/11509

## Checklist

 - [x] Added an entry in `CHANGES.txt` for user facing changes
 - [x] Updated documentation & `sql_features` table for user facing changes
 - [x] Touched code is covered by tests
 - [x] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [x] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -42,6 +42,10 @@ Unreleased Changes
 Breaking Changes
 ================
 
+- Changed the type of :ref:`TIMESTAMP <type-timestamp>` from :ref:`TIMESTAMP WITH
+  TIME ZONE <type-timestamp-with-tz>` to :ref:`TIMESTAMP WITHOUT TIME ZONE
+  <type-timestamp-without-tz>` to be compatible with PostgreSQL.
```
Comment (seut) [inline docs/appendices/release-notes/unreleased.rst:47]:
```suggestion
  <type-timestamp-without-tz>` to be compatible with the SQL standard.
```

File: docs/general/ddl/data-types.rst [hunk 2]
```diff
@@ -1121,6 +1121,15 @@ With a few exceptions (noted below), the ``+`` and ``-`` :ref:`operators
 A timestamp expresses a specific date and time as the number of milliseconds
 since the `Unix epoch`_ (i.e., ``1970-01-01T00:00:00Z``).
 
+Timestamp has two variants:
```
Comment (seut) [inline docs/general/ddl/data-types.rst:1124]:
```suggestion
``TIMESTAMP`` has two variants:
```

File: docs/general/ddl/data-types.rst [hunk 3]
```diff
@@ -1121,6 +1121,15 @@ With a few exceptions (noted below), the ``+`` and ``-`` :ref:`operators
...
+    - :ref:`TIMESTAMP WITHOUT TIME ZONE <type-timestamp-without-tz>` which
+       presents all values in UTC.
+    - :ref:`TIMESTAMP WITH TIME ZONE <type-timestamp-with-tz>`  which presents
+      all values in UTC in respect to the TIME ZONE related offset.
+
+By default a Timestamp is an alias for :ref:`TIMESTAMP WITHOUT TIME ZONE
```
Comment (seut) [inline docs/general/ddl/data-types.rst:1130]:
```suggestion
By default a ``TIMESTAMP`` is an alias for :ref:`TIMESTAMP WITHOUT TIME ZONE
```

File: docs/general/ddl/data-types.rst [hunk 4]
```diff
@@ -1281,13 +1290,7 @@ timestamp into UTC prior to insertion. Contrast this with the behavior of
     In the absence of an explicit :ref:`WITH TIME ZONE
     <type-timestamp-with-tz>` or :ref:`WITHOUT TIME ZONE
     <type-timestamp-without-tz>`, CrateDB will interpret ``TIMEZONE`` as an
```
Comment (seut) [inline docs/general/ddl/data-types.rst:1292]: Should this phrase ``TIMESTAMP`` instead of ``TIMEZONE``?

File: docs/general/ddl/data-types.rst [hunk 5]
```diff
@@ -1281,13 +1290,7 @@ timestamp into UTC prior to insertion. Contrast this with the behavior of
...
-    alias for ``TIMESTAMP WITH TIME ZONE``.
-
-    This behaviour does not comply with standard SQL and is incompatible with
-    PostgreSQL. CrateDB 4.0 :ref:`deprecated this alias <v4.0.0-deprecations>`
-    and behavior may change in a future version of CrateDB (see `tracking issue
-    #11491`_). To avoid issues, we recommend that you always specify ``WITH
-    TIME ZONE`` or ``WITHOUT TIME ZONE``.
+    alias for ``TIMESTAMP WITHOUT TIME ZONE``.
```
Comment (seut) [inline docs/general/ddl/data-types.rst:1293]: Shall we maybe remove the whole `CAUTION` paragraph now that it behaves as the SQL standard defines?

File: docs/general/ddl/data-types.rst [hunk 6]
```diff
@@ -1352,14 +1355,7 @@ literal. Contrast this with the behavior of :ref:`WITH TIME ZONE
...
     <type-timestamp-without-tz>`, CrateDB will interpret ``TIMEZONE`` as an
-    alias for ``TIMESTAMP WITH TIME ZONE``.
-
-    This behaviour does not comply with standard SQL and is incompatible with
-    PostgreSQL. CrateDB 4.0 :ref:`deprecated this alias <v4.0.0-deprecations>`
-    and behavior may change in a future version of CrateDB (see `tracking issue
-    #11491`_). To avoid issues, we recommend that you always specify ``WITH
-    TIME ZONE`` or ``WITHOUT TIME ZONE``.
-
+    alias for ``TIMESTAMP WITHOUT TIME ZONE``.
```
Comment (seut) [inline docs/general/ddl/data-types.rst:1358]: same here, I think we should remove this paragraph.

File: docs/general/ddl/data-types.rst (hunk 4 above)
Comment (mkleen) [reply docs/general/ddl/data-types.rst:1292]: yes, you are right.

File: docs/general/ddl/data-types.rst (hunk 5 above)
Comment (mkleen) [reply docs/general/ddl/data-types.rst:1293]: Sounds good to me.

**Review by seut — APPROVED** [review]
👍

**Review by mfussenegger — COMMENTED** [review]
I think this can be removed as well:

https://github.com/crate/crate/blob/ffba331c0f012b8143ad233fe2b036938730b889/server/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java#L276-L282

**Comment by mkleen** [comment]
> I think this can be removed as well:
> 
> https://github.com/crate/crate/blob/ffba331c0f012b8143ad233fe2b036938730b889/server/src/main/java/io/crate/analyze/AnalyzedColumnDefinition.java#L276-L282

Good catch, thank you!

**Pull request merged**